contracts/contracts/Umbra.sol

```diff
@@ -1,10 +1,63 @@
 pragma solidity ^0.5.0;
 
 import "@openzeppelin/contracts-ethereum-package/contracts/ownership/Ownable.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20.sol";
 
-contract Umbra is Ownable {
+contract Umbra is Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
 
-    constructor() public {
+    event EthAnnouncement(address indexed receiver, uint256 indexed amount, string note);
```

Comment:
We might be able to change `string` to `bytes` which should save some gas, but we can wait until we have actual encrypted payloads to test with to ensure that works